### PR TITLE
updated for ubid, remove address and city

### PR DESCRIPTION
**Pull request by Mr-Mickey**
Removed address and city to support half addresses with ubid

### CODE REVIEW

**Comment by charliepnnl** [comment]
Don't we need some logic that says if the address or city are NOT specific, then the ubid must be specified?

**Review by yan130 — APPROVED** [review]

**Comment by thepriestscottage** [comment]
This is not complete.  You need to add methods that check that if these are missing then the user has provided coordinates.

File: home_audit.node.js [hunk 1]
```diff
@@ -1605,8 +1627,11 @@ function validate_home_audit (homeValues, additionalRules = null) {
  */
 function validate_address (homeValues) {
     let mandatoryMessage = "Missing value for mandatory field";
+    let optionalMessage = "Required field, if no cooridnates are provided."
```
Comment (thepriestscottage) [inline home_audit.node.js]: Incorrect spelling of "coordinates."  The grammar here is also weird, so I would recommended: "Field is required if no coordinates are provided."

File: home_audit.node.js [hunk 2]
```diff
@@ -466,6 +466,7 @@ let validationRules = {
         return new Validation(TypeRules._string(value), ERROR);
     },
     city: function(value) {
+
```
Comment (thepriestscottage) [inline home_audit.node.js:469]: Can we remove this extra line? :)

**Review by thepriestscottage — APPROVED** [review]
A couple thoughts, but this should be good!

File: home_audit.node.js [hunk 3]
```diff
@@ -1533,11 +1534,30 @@ let validationRules = {
                 `;
             }
         }
+    },
+
+    /*
+     * Checks that if coordinates or address has been added.
```
Comment (charliepnnl) [inline home_audit.node.js:1540]: Can you reword the description of this function? Maybe something like:
```
Treats a field as required when coordinates are not provided.

Intended to be used for the address and city fields.
```

File: home_audit.node.js [hunk 4]
```diff
@@ -1533,11 +1534,30 @@ let validationRules = {
...
+        let latHi =  _homeValues.latitude_hi;
+        let longHi = _homeValues.longitude_hi;
+        let latCen = _homeValues.latitude_center;
+        let longCen = _homeValues.longitude_center
+        
+        if((latLo && longLo && latHi && longHi && latCen && longCen) || value){
+            return null;
+        }
+        else{
+            return "Field is required if no cooridnates are provided."
```
Comment (charliepnnl) [inline home_audit.node.js:1555]: `coordinates` is misspelled.

**Pull request merged**